transcom/mymove
# [MB-6430] Manual date entry in DatePickerInput on TIO filter triggers 500 error

## Description

Editing the Submitted at date filter in the TIO payment requests queue manually with the keyboard instead of using the Date Picker UI would cause an 'Invalid date' value being sent to the api causing a 500 error.  The field handled invalid or incomplete dates properly, it was only when you provided a valid date in the format 'DD MMM YYYY' that it sent the bad request.

I've updated the swagger param for the submittedAt filter to be of format date so that it will catch bad requests and return a 422 error instead of a 500 error.

For the DatePickerInput component I saw we were reusing the onChange handler for both when the SingleDateInput picker UI is updated as well as part of the inputProps being passed the input field itself.  I removed attaching the onChange listener to the underlying `<input>` form field.

This DatePickerInput is being used on the Customer side as well in the Orders form and I saw the validation message there was not a user friendly error message so I updated the copy based on recommendations from James https://ustcdp3.slack.com/archives/CTNTFJSBA/p1611766050125800?thread_ts=1611684009.111700&cid=CTNTFJSBA

## Reviewer Notes

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make office_client_run
```

Test TIO Queue Filter
1. Login as the TIO user
2. Type a date into the Submitted queue filter text input in the format of '26 Jan 2020'
3. The queue should properly filter the payment requests sending the proper value to the api '2020-01-26' instead of 'Invalid date' causing a 500 error

Test New Validation Message
1. Login as the TOO/TIO user and then navigate to the Edit Orders page either from the Move Details or Move Payment Request page
2. For orders issue date or report by date, type in a malformatted date such as '2021-31-01' or incomplete date '27 Jan 20' and move focus from the input to trigger the form validation

1. Login as a Service Member needing orders information such as the needs@orde.rs test user
2. Choose add orders from the home screen
3. For orders issue date or report by date, type in a malformatted date such as '2021-31-01' or incomplete date '27 Jan 20' and move focus from the input to trigger the form validation

## Code Review Verification Steps


 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-6430) for this change

## Screenshots

https://user-images.githubusercontent.com/52669884/105884793-5645ff00-5fd6-11eb-8efa-a85da92d5794.mov

https://user-images.githubusercontent.com/52669884/106014502-8c918600-608b-11eb-8bb7-7b3f5a6f8659.mov

Previous Service Member orders validation message
<img width="403" alt="Screen Shot 2021-01-26 at 12 24 02 PM" src="https://user-images.githubusercontent.com/52669884/106019360-8fdb4080-6090-11eb-8a3c-1f4cb125ef1a.png">

Updated Service Member orders validation message
![Screen Shot 2021-01-27 at 2 25 32 PM](https://user-images.githubusercontent.com/52669884/106043443-6defb700-60ac-11eb-985f-3f069639bb55.png)

Updated TXO edit orders form validation message
![Screen Shot 2021-01-27 at 2 24 08 PM](https://user-images.githubusercontent.com/52669884/106043479-7d6f0000-60ac-11eb-8db3-194dd2976aac.png)
